### ISSUE TITLE
Debugging type `chrome` fails when using `request: attach` in a compound.

### ISSUE THREAD

**Issue by CynicalBusiness** [microsoft/vscode-js-debug#775]
**Describe the bug**
When using `"type": "chrome"` and `"request": "attach"`, the debugger will fail to start almost immediately with the following message: 
```
Cannot connect to the target at localhost:9222: Could not connect to debug target at http://localhost:9222: Promise was canceled.
```
The port in the message will change depending on what the debug task's `port` option is set to. Setting `timeout` has no affect on this.

If I start the individual task after the Chrome process has started completely, it works fine. Only when run as a compound do I see this error.

**To Reproduce**
1. Set up a launch task with `"type": "chrome"` and `"request": "attach"`.
2. Set up some other launch task. In my case this is an electron main process with the `--remote-debugging-port` arg.
3. Put the two tasks into a compound
4. Try to run the compound, see error after a few seconds.

**Log File**

[vscode-debugadapter-0.json.gz](https://github.com/microsoft/vscode-js-debug/files/5295894/vscode-debugadapter-0.json.gz)

**VS Code Version:** 1.49.1

**Additional context**
My `launch.json` is the following:
```json
{
    "version": "0.2.0",
    "configurations": [
        {
            "name": "App:Main",
            "type": "node",
            "request": "launch",
            "cwd": "${workspaceRoot}/packages/app",
            "runtimeExecutable": "${workspaceRoot}/node_modules/.bin/electron",
            "windows": {
                "runtimeExecutable": "${workspaceRoot}/node_modules/.bin/electron.cmd"
            },
            "preLaunchTask": "gulp: rebuild/client",
            "autoAttachChildProcesses": true,
            "outputCapture": "std",
            "runtimeArgs": ["--remote-debugging-port=9333"],
            "args": ["./lib/main.js", "--enableDebugging"],
            "console": "internalConsole"
        },
        {
            "name": "App:Client",
            "type": "chrome",
            "request": "attach",
            "port": 9333,
            "timeout": 30000,
            "webRoot": "${workspaceRoot}",
            "sourceMaps": true,
            "sourceMapPathOverrides": {
                "webpack:///./~/*": "${workspaceRoot}/node_modules",
                "webpack:///./*": "${workspaceRoot}/*",
                "webpack:///*": "*"
            }
        }
    ],
    "compounds": [
        {
            "name": "App",
            "configurations": ["App:Main", "App:Client"]
        }
    ]
}

```

**Comment by connor4312** [comment]
Hey, apologies for the delayed response on this.

The error message you posted references failing to connect ton port 9222, but you use 9333 in your launch config. Is the error or your launch config mismatched?

In the logs you posted, it looks like we continue to get `ECONNREFUSED 127.0.0.1:9333`, which means it doesn't look like the renderer is listening on that port.

**Comment by CynicalBusiness** [comment]
Sorry for the confusion on that; I must have copied the error when testing if other ports worked. The error appears with the port provided regardless of what it is. It does indeed connect if I simply run the chrome attach task separately after the renderer has visibly opened and works fine, just not as part of the compound.

Another interesting note: I noticed on a few rare builds it does actually work, so it smells like timeout issues or a race condition to me (maybe with the pre-launch task?), but I do not know for sure.

**Comment by connor4312** [comment]
Hm, okay, thanks. Two things I wanted to confirm:

1. You said "the debugger will fail to start almost immediately with the following message". In the log you sent (and generally what I would expect) is that it only shows that failure when the timeout is reached, after 30s in this case.

1. In this scenario, does either `http://localhost:9333/json/version` or `http://localhost:9333/json/list` work? We expect the debug connection information to be published on either of those two URLs.

**Comment by CynicalBusiness** [comment]
The message was a second or so after pressing "Start Debugging," more accurately, regardless of timeout. It was timed right about the moment when the terminal for the pre-launch task appeared.

Interestingly, though, I went to make some changes to my build scripts (namely the one in the pre-launch task of the main process), and it quit doing it. Now it's gone back to the expected behavior of timing out after 30s when my pre-launch is longer than that.

The way it was acting kinda seems like the pre-launch task itself was getting picked up by the debugger? VSCode has been doing some strange things ever since the auto-attach was updated in 49, like auto-attaching to yarn cli, so I would not be surprised.

Regardless, I think it was more a problem on VSCode's side or my build script at this point. It is really strange.

**Comment by connor4312** [comment]
Auto attach could be attaching to your launch.json tasks. Its "smart" mode is much improved in 1.50 (which will be released this week) so it should no longer attach to yarn, among other things. I don't _think_ it's causing the error you're seeing, since it will listen on port 0 (any free port) by default, though the `--enableDebugging` flag you're passing raises my suspicions that something more might be happening...

> Now it's gone back to the expected behavior of timing out after 30s when my pre-launch is longer than that.

Is it now working for you / working when your pre-launch is < 30 seconds long?

**Comment by CynicalBusiness** [comment]
> Is it now working for you / working when your pre-launch is < 30 seconds long?

Correct: the timeout length is being respected again, along with the message actually saying "connection timed out" (something along those lines) in the popup now. Over the week, between continuing to work on the project, messing with the build scripts, and VSCode updating from 1.49.0 to 1.49.3, it appears to have resolved itself. Not sure why though.

For reference, `--enableDebugging` is an argument the script itself takes for debug-level logging and enabling the chrome devTools: AFAIK it doesn't do anything else, but maybe something is picking it up too. Removing it had no effect before.

That said, I think it can be closed for now unless somebody else sees the issue.

**Comment by connor4312** [comment]
Ok, thanks. I think there were some fixes between 1.49.0->3, and as mentioned there will be more coming later this week.

Thanks for opening an issue and helping debug it!
